### PR TITLE
BZ2089824: Deleting a baremetalhost order

### PR DESCRIPTION
**Pull request by amolnar-gh**
Version(s):
4.8, 4.9, 4.10, 4.11

Issue:
https://bugzilla.redhat.com/show_bug.cgi?id=2089824

### CODE REVIEW

**Comment by e-minguez** [comment]
LGTM!

**Comment by e-minguez** [comment]
cc @zaneb @dhellmann @dtantsur

**Comment by dhellmann** [comment]
These changes look right to me, thanks @e-minguez !

**Comment by ptrnull** [comment]
/lgtm
Can we get this merged asap?

**Comment by amolnar-rh** [comment]
@ptrnull I requested a review and a merge an hour ago. I don't have an exact ETA, but should be done by end of the day the latest.

File: modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc [hunk 1]
```diff
@@ -17,15 +17,16 @@ Existing control plane `BareMetalHost` objects may have the `externallyProvision
...
+. Remove the old `baremetalhost` and `machine` objects:
 +
 [source,terminal]
 ----
-$ oc delete node kni1-vmaster-0
 $ oc delete bmh -n openshift-machine-api vmaster-0
 $ oc delete machine -n openshift-machine-api kni1-master-0
 ----
 +
+After removing the `baremetalhost` and `machine` objects, then the baremetal-operator deletes the `node` object automatically.
```
Comment (darshan-nagaraj) [inline modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc:28]: Just a suggestion. wdyt ?
```suggestion
After you remove the `baremetalhost` and `machine` objects, then the baremetal-operator deletes the `node` object automatically.
```

Comment (pneedle-rh) [inline modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc:28]: You could also change ", then the baremetal-operator deletes" to the following:

> , the Bare Metal Operator deletes...

File: modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc [hunk 2]
```diff
@@ -17,15 +17,16 @@ Existing control plane `BareMetalHost` objects may have the `externallyProvision
 
 .Procedure
 
-. Remove the old `baremetalhost`, `machine` and `node` objects:
+. Remove the old `baremetalhost` and `machine` objects:
```
Comment (pneedle-rh) [inline modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc:20]: In the important admonition at the start of this module, the "`BareMetalHost` object" is referred to. You could either change your "`baremetal` object" references throughout this PR to use the same capitalisation as in the admonition, or vice-versa. I recommend reviewing which of the two is the correct capitalisation for the object name. My guess is that it's "`BareMetalHost`", based on the YAML in step 2.

**Review by pneedle-rh — COMMENTED** [review]
@amolnar-rh I have now completed my review of this PR. Please let me know if you have any questions about my suggestions. Thanks!

File: modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc (hunk 1 above)
Comment (amolnar-rh) [reply modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc:28]: Updated :)

File: modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc (hunk 2 above)
Comment (amolnar-rh) [reply modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc:20]: @pneedle-rh You're right and it seems that `machine` should be `Machine` as well. Updated accordingly

Comment (pneedle-rh) [reply modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc]: Thank you for highlighting that. Please also review if `node` should be `Node` when referring to the object.

Comment (pneedle-rh) [reply modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc]: Looking through the docs library, the `Node` object is capitalised too.

Comment (amolnar-rh) [reply modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc]: Thank you, Paul!

**Comment by pneedle-rh** [comment]
LGTM

File: modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc [hunk 3]
```diff
@@ -17,16 +17,17 @@ Existing control plane `BareMetalHost` objects may have the `externallyProvision
...
 +
 [source,terminal]
 ----
-$ oc delete node kni1-vmaster-0
 $ oc delete bmh -n openshift-machine-api vmaster-0
 $ oc delete machine -n openshift-machine-api kni1-master-0
 ----
 +
-. Create the new `baremetalhost` object and the secret to store the BMC credentials:
+After you remove the `BareMetalHost` and `Machine` objects, then the Bare Metal Operator automatically deletes the `Node` object.
```
Comment (zaneb) [inline modules/ipi-install-replacing-a-bare-metal-control-plane-node.adoc:28]: FWIW it is the Machine controller and not the baremetal-operator that deletes the Node.

**Comment by amolnar-rh** [comment]
[This PR](https://github.com/openshift/openshift-docs/pull/46107) addresses Zane Bitter's comments.

**Pull request merged**